core/server/master/src/main/java/alluxio/master/DefaultAlluxioMaster.java

```diff
@@ -130,7 +135,8 @@ protected DefaultAlluxioMaster() {
       }
       mTransportProvider = TransportProvider.Factory.create();
       mTServerSocket =
-          new TServerSocket(NetworkAddressUtils.getBindAddress(ServiceType.MASTER_RPC));
+          new TServerSocket(NetworkAddressUtils.getBindAddress(ServiceType.MASTER_RPC),
+                  connectionTimeout);
```

Comment:
I think this is a good knob to have. I have two concerns of this approach:
1.  This can only work for TThreadPool-like sever. If we ever want to change to other type of servers and still want to gc connections, this won't work.
2. This prevents the client from keeping a minimum amount of connections.

The other approach suggested in https://alluxio.atlassian.net/browse/ALLUXIO-2555 doesn't have these problems. It is the approach being used by all the worker connections (thrift and netty) in Alluxio now. But for now, I think it is ok to merge this PR to solve your problem.